### PR TITLE
chore: improve CMake builds on macOS and Linux

### PR DESCRIPTION
**Pull request by erickzhao**
## Description
This PR aims to make CMake build the actual function app by globbing all files in `core-linux/**/*.cpp` and `core-macOS/**/*.cpp`.

## Changes proposed
- Globs additional files.
- Puts out fires to make everything compile. Note that I moved some stuff out of `core-shared` into `core-linux`, so this will probably break the Windows code. We'll have to copy/paste that manually for now.

## How to test it

On macOS/Linux:
```sh
$ cd build
$ cmake ..
$ make
```

The `neutralino` binary should be produced. Running it directly in the `build` folder should output an error about a missing `settings.json`. To provide a minimal environment for the binary, download the latest Neutralino release and drop the binary in that folder. It should run:

image

On Linux Docker Container:
```sh
$ docker build -t . test
$ docker run test
```

So far, I haven't actually tested the output binary. However, linking all of the missing `.cpp` files seems like the right thing to do...

## Next steps
Do the same for Windows.

## Deploy notes

None.

### CODE REVIEW

**Comment by kraglalbert** [comment]
Looks good I think, but can you add the steps to test these changes?

File: cmake-utils/main.cc [hunk 1]
```diff
@@ -1,7 +1,7 @@
-#include <iostream>
-#include <string>
+// #include <iostream>
```
Comment (erickzhao) [inline cmake-utils/main.cc:1]: This conflicts with the `main.cpp` from `core-*`

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,35 +7,37 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
     message(STATUS "Linux Detected")
+
+    # Use the package PkgConfig to detect GTK+ headers/library files
     find_package(PkgConfig REQUIRED)
-	pkg_check_modules(GTK3 REQUIRED gtk+-3.0)
+    pkg_check_modules(GTK3 REQUIRED gtk+-3.0)
+    pkg_check_modules(WEBKIT2GTK REQUIRED webkit2gtk-4.0)
 	pkg_check_modules(WEBKIT REQUIRED webkitgtk-3.0)
+
+    # Setup CMake to use GTK+ and Webkit2GTK, tell the compiler where to
```
Comment (erickzhao) [inline CMakeLists.txt:45]: Moved your GTK code to the Linux-specific parts

**Review by AndreiUngur — DISMISSED** [review]
Worked locally as long as the contents of `dist` were in the same folder as my executable. Steps to reproduce:

```
docker build -t . test
docker run -v `pwd`/build:/neutralinojs/build test
cd build
./neutralino
```
Assuming `build` contains the same stuff as `dist` :tada:

**Review by kraglalbert — DISMISSED** [review]
👍

**Review by AndreiUngur — APPROVED** [review]
Still LGTM after rebase

**Review by kraglalbert — APPROVED** [review]

**Pull request merged**